scripts/experiment_2/nbconverted/4_similarity_analysis_cca.py

```diff
@@ -0,0 +1,157 @@
+
+# In[1]:
+
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+import os
+import ast
+import pandas as pd
+import numpy as np
+import random
+import glob
+from plotnine import *
+from sklearn.cross_decomposition import CCA
+import warnings
+warnings.filterwarnings(action='ignore')
+
+
+# In[2]:
+
+
+# Load config file
+config_file = "config_exp_2.txt"
+
+d = {}
+float_params = ["learning_rate", "kappa", "epsilon_std"]
+str_params = ["analysis_name", "NN_architecture"]
+lst_params = ["num_batches"]
+with open(config_file) as f:
+    for line in f:
+        (name, val) = line.split()
+        if name in float_params:
+            d[name] = float(val)
+        elif name in str_params:
+            d[name] = str(val)
+        elif name in lst_params:
+            d[name] = ast.literal_eval(val)
+        else:
+            d[name] = int(val)
+
+
+# In[3]:
+
+
+# Parameters
+analysis_name = d["analysis_name"]
+NN_architecture = d["NN_architecture"]
+num_batches = d["num_batches"]
+
+
+# In[4]:
+
+
+# Load data
+base_dir = os.path.abspath(os.path.join(os.getcwd(),"../.."))
+
+simulated_data_file = os.path.join(
+    base_dir,
+    "data",
+    "simulated",
+    analysis_name,
+    "simulated_data.txt.xz")
+
+batch_dir = os.path.join(
+    base_dir,
+    "data",
+    "batch_simulated",
+    analysis_name)
+
+
+# In[5]:
+
+
+# Read in data
+simulated_data = pd.read_table(
+    simulated_data_file,
+    header=0, 
+    index_col=0,
+    sep='\t')
+
+simulated_data.head(10)
+
+
+# In[6]:
+
+
+get_ipython().run_cell_magic('time', '', '# Calculate similarity using CCA\ncca = CCA(n_components=10)\noutput_list = []\n\nfor i in num_batches:\n    print(\'Cacluating CCA of 1 batch vs {} batches..\'.format(i))\n    \n    # Get batch 1 data\n    batch_1_file = os.path.join(\n        batch_dir,\n        "Batch_1.txt.xz")\n\n    batch_1 = pd.read_table(\n        batch_1_file,\n        header=0,\n        index_col=0,\n        sep=\'\\t\')\n\n    # Simulated data with all samples in a single batch\n    original_data_df =  batch_1\n    \n    # Get data with additional batch effects added\n    batch_other_file = os.path.join(\n        batch_dir,\n        "Batch_"+str(i)+".txt.xz")\n\n    batch_other = pd.read_table(\n        batch_other_file,\n        header=0,\n        index_col=0,\n        sep=\'\\t\')\n    \n    # Simulated data with i batch effects\n    batch_data_df =  batch_other\n    \n    # CCA\n    U_c, V_c = cca.fit_transform(original_data_df, batch_data_df)\n    result = np.mean(np.corrcoef(U_c.T, V_c.T)) ## TOP singular value or mean singular value???\n    \n    output_list.append(result)')
```

Comment:
conversion failed again

Reply:
why not get a matrix norm instead of mean or top corrcoef?

Reply:
This is something I started to think about.  For now I am taking the mean to be able to have a fair comparison against svcca.